tests/test_data_processor.py

```
@@ -0,0 +1,42 @@
+    data_processor.preprocess_data()
+    X_train, X_test, y_train, y_test = data_processor.split_data(test_size=0.4, random_state=42)
+    
+    assert X_train.shape == (3, 4)
+    assert X_test.shape == (2, 4)
+    assert y_train.shape == (3,)
+    assert y_test.shape == (2,)
+
+def test_preprocessor_transform(data_processor):
+    data_processor.preprocess_data()
+    X_transformed = data_processor.preprocessor.fit_transform(data_processor.X)
+    
```

Ruff

Blank line contains whitespace

Remove whitespace from blank line

(W293)